src/transformers/models/sam/modeling_sam.py

```diff
@@ -755,9 +771,9 @@ def get_rel_pos(self, q_size: int, k_size: int, rel_pos: torch.Tensor) -> torch.
             rel_pos_resized = rel_pos
 
         # Scale the coords with short length if shapes for q and k are different.
-        q_coords = torch.arange(q_size)[:, None] * max(k_size / q_size, 1.0)
-        k_coords = torch.arange(k_size)[None, :] * max(q_size / k_size, 1.0)
-        relative_coords = (q_coords - k_coords) + (k_size - 1) * max(q_size / k_size, 1.0)
+        q_coords = torch.arange(q_size)[:, None] * k_size / q_size
+        k_coords = torch.arange(k_size)[None, :] * q_size / k_size
+        relative_coords = (q_coords - k_coords) + (k_size - 1) * q_size / k_size
```

Comment:
This change is bad, but the question is: are `k_size` and `q_size` known at tracing-time?